glean_parser/util.py

```diff
@@ -66,13 +66,40 @@ def remove_implicit_resolver(cls, tag_to_remove):
 _NoDatesSafeLoader.remove_implicit_resolver("tag:yaml.org,2002:timestamp")
 
 
+class SafeLineLoader(_NoDatesSafeLoader):
+    """
+    Map line number to ping and metric yaml nodes.
+
+    There are a few specific places we can map the line number to:
+    only add line number to ping/metric node, and map it at the
+    *property* level so that it won’t fail schema validation.
+    Since the file structures for pings.yaml and metrics.yaml
+    are different, and yamllint is run on the input which enforces
+    2-space tab width, the logic to find the right nodes to map
+    to is defined in construct_mapping().
+    """
+
+    def construct_mapping(self, node, deep=False):
+        mapping = super(SafeLineLoader, self).construct_mapping(node, deep=deep)
+
+        for value in node.value:
+            if value[0].value != "$schema":
+                # pings.yaml
+                if node.start_mark.column == 2 and value[0].value == "description":
+                    mapping["defined_in"] = {"line": str(node.start_mark.line)}
+                # metrics.yaml
+                if node.start_mark.column == 4 and value[0].value == "type":
+                    mapping["defined_in"] = {"line": str(node.start_mark.line)}
+        return mapping
+
+
 if sys.version_info < (3, 7):
     # In Python prior to 3.7, dictionary order is not preserved. However, we
     # want the metrics to appear in the output in the same order as they are in
     # the metrics.yaml file, so on earlier versions of Python we must use an
     # OrderedDict object.
     def ordered_yaml_load(stream):
-        class OrderedLoader(_NoDatesSafeLoader):
+        class OrderedLoader(SafeLineLoader):
```

Comment:
This works fine for other Python versions, but for some reason OrderedLoader is not instantiated and used correctly in v3.6 which might have caused the failed build-36 test. I was wondering what you think is the best way to fix this. Thanks @mdboom